evm_trace/gas.py

```diff
@@ -0,0 +1,54 @@
+from typing import Dict, List
+
+from evm_trace.base import CallTreeNode
+
+Address = str
+Calldata = bytes
+GasUsage = int
+
+GasReport = Dict[Address, Dict[Calldata, List[GasUsage]]]
+
+
+def extract_gas_report(calltree: CallTreeNode) -> GasReport:
```

Comment:
Just thinking out loud here, might be a nicer way to do the filtering for `method_id` then having to do another passing at combining when used inside of ape (but still keeping it general)
```suggestion
def extract_gas_report(calltree: CallTreeNode, calldata_filter_fn: Callable[[bytes], bytes]=lambda d: d) -> GasReport:
```

Reply:
Maybe could even have a special function that reduces all calldata to `b""` and therefore flattens the multilevel dict structure to just `Dict[Address, List[GasUage]]`, but that seems out of scope for now